Bearded.Utilities/Monads/Result.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Bearded.Utilities.Monads
+{
+    public readonly struct Result<TResult, TError> : IEquatable<Result<TResult, TError>>
+    {
+        private readonly bool isSuccess;
+        private readonly TResult result;
+        private readonly TError error;
+
+        private Result(bool isSuccess, TResult result, TError error)
```

Comment:
Do I see correctly that the only way of construction is via the Success and Failure casts?

Reply:
Maybe<T> has public construction methods on the type itself. I think we should perhaps have them here as well, so people don't have to use the other types? Or do you think we should enforce the 'nice' way (really the way that we think is nicest)?

Reply:
Fair enough, added.